packages/amazonq/src/extensionShared.ts

```diff
@@ -36,6 +37,20 @@ export async function activateShared(context: vscode.ExtensionContext) {
     const contextPrefix = 'amazonq'
     globals.contextPrefix = 'amazonq.' //todo: disconnect from above line
 
+    // Avoid activation if older toolkit is installed
+    // Amazon Q is only compatible with AWS Toolkit >= 3.0.0
+    const toolkit = vscode.extensions.getExtension(VSCODE_EXTENSION_ID.awstoolkit)
+    if (toolkit) {
+        const toolkitVersion = semver.coerce(toolkit.packageJSON.version)
+        if (toolkitVersion && toolkitVersion.major < 3) {
```

Comment:
nit: We can make this shorter by using optional chaining operator.

```suggestion
        if (toolkitVersion?.major < 3) {
```

Reply:
<img width="597" alt="Screenshot 2024-04-15 at 3 42 49 PM" src="https://github.com/aws/aws-toolkit-vscode/assets/97199248/7e817c6a-73fb-45f2-8463-16267108afaf">

Reply:
The optional chaining should've work and something feels a little weird here.

Verified om typescript lang - Typescript compiles `toolkitVersion?.major < 3` to `(toolkitVersion === null || toolkitVersion === void 0 ? void 0 : toolkitVersion.major) < 3;`

https://www.typescriptlang.org/play?#code/C4exBsGsEtgNQKYCcDO0QDsD8A6AtgIYBWISABADxkDMQA